### ISSUE TITLE
errors in latest release

### ISSUE THREAD

**Issue by psifertex** [toolCHAINZ/hexfiles#5]
```
Traceback (most recent call last):
  File "/Applications/Binary Ninja.app/Contents/MacOS/plugins/../../Resources/python/binaryninja/binaryview.py", line 2237, in _is_valid_for_data
    return cls.is_valid_for_data(BinaryView(handle=core.BNNewViewReference(data)))  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/$user/Library/Application Support/Binary Ninja/repositories/community/plugins/toolCHAINZ_hexfiles/src/base_view.py", line 24, in is_valid_for_data
    return is_valid_for_data(data, cls.hexfile_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/#user/Library/Application Support/Binary Ninja/repositories/community/plugins/toolCHAINZ_hexfiles/src/helper.py", line 35, in is_valid_for_data
    return is_srec(trimmed.decode("utf8"))  # type: ignore
                   ^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte
```

TL;DR non-text file input are harmful to attempting to decode as utf8. 

Possible solutions, from best to worst:
- Given that .hex files are supposed to only contain ascii, maybe just pre-processing the file to return False if anything is non-ascii would be safe?
- Or just use "ascii" for decode instead of "utf8"? False positives seem like they would be low that way.
- To something terrible like wrap it in a try/accept

**Comment by toolCHAINZ** [comment]
Hmmm, good catch; I guess I should add some unit tests for that. I'll try some stuff and make a new release.

**Comment by toolCHAINZ** [comment]
This should be fixed in 1.2.0. Let me know if I missed something.

**Comment by psifertex** [comment]
looks good to me, thanks!
